### ISSUE TITLE
Syntax error, unexpected T_MATCH on line

### ISSUE THREAD

**Issue by Danon** [nikic/PHP-Parser#715]
I've tried to parse PHP code, and I've been getting this error:
```
Syntax error, unexpected T_MATCH on line
```

I've prepared a PhpUnit test, just paste it in any project with https://github.com/nikic/PHP-Parser and run it. You will see that the first two data providers fail, while the rest passes. For some reason signature `Match $match` isn't valid. 

I'm sure it is though, since I use it extensively in my [T-Regx](https://t-regx.com/) library, and it works fine.

```php
<?php
namespace Test\Bugs;

use PhpParser\Error;
use PhpParser\ParserFactory;
use PHPUnit\Framework\TestCase;

class Test extends TestCase
{
    /**
     * @test
     * @dataProvider codes
     * @param string $code
     */
    public function shouldNotInfluenceFurtherChecks(string $code)
    {
        // given
        $parser = (new ParserFactory)->create(ParserFactory::ONLY_PHP7);

        // when
        try {
            $parser->parse($code);

            // then
            $this->assertTrue(true);
        } catch (Error $error) {
            $this->assertTrue(false, $error->getMessage());
        }
    }

    public function codes(): array
    {
        return [
            ['<?php function test(Match $match) {}'],  // this fails
            ['<?php function test(Match $matchXX) {}'],  // this fails
            ['<?php function test(Match $xxMatch) {}'],  // the rest passes
            ['<?php function test(Match $param) {}'],

            ['<?php function test(MatchXX $match) {}'],
            ['<?php function test(MatchXX $matchXX) {}'],
            ['<?php function test(MatchXX $xxMatch) {}'],
            ['<?php function test(MatchXX $param) {}'],
        ];
    }
}
```

**Comment by nikic** [comment]
The ParserFactory will use forward-emulation by default, which means that it will parse your code as PHP 8. Your code is illegal in PHP 8.

You can use `$parser = (new ParserFactory)->create(ParserFactory::ONLY_PHP7, new Lexer());` if you want to use the PHP version you are running on.

**Comment by Danon** [comment]
@nikic But I used  `create(ParserFactory::ONLY_PHP7);`. Isn't that enough? Do I have to add `new Lexer()`?

**Comment by kelunik** [comment]
@Danon `match` is a new keyword I guess.

**Comment by nikic** [comment]
@Danon Oh, good point. The first four should all be illegal, but due to a bug, only those that contain a lowercase `match` somewhere are...

**Comment by nikic** [comment]
The case sensitivity issue should be fixed by b5351f883a62fd85e4fbaafc3a0b7035bc28fab1.

**Comment by Danon** [comment]
Is the issue fixed now?
